CMSIS/NN/Source/NNSupportFunctions/arm_nn_mat_mul_core_4x_s8.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2010-2020 Arm Limited or its affiliates. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* ----------------------------------------------------------------------
+ * Project:      CMSIS NN Library
+ * Title:        arm_nn_mat_mul_core_4x_s8.c
+ * Description:  General matrix multiplication function for MVE extension
+ *
+ * $Date:        January 15, 2020
+ * $Revision:    V.1.0.0
+ *
+ * Target Processor:  Cortex-M cores
+ * -------------------------------------------------------------------- */
+
+#include "arm_math.h"
+#include "arm_nnfunctions.h"
+
+/**
+ * @ingroup groupSupport
+ */
+
+/**
+ * @addtogroup NNBasicMath
+ * @{
+ */
+
+/*
+   * s8 matrix multiplication to process 4 rows and one column
+   *
+   * Refer header file for details.
+   *
+   */
+arm_status arm_nn_mat_mul_core_4x_s8(const int32_t row_elements,
+                                     const int8_t *row_base,
+                                     const int8_t *col_base,
+                                     int32_t *const sum_col,
+                                     int32_t *const output)
+{
+    int32_t acc_n0 = 0;
+    int32_t acc_n1 = 0;
+    int32_t acc_n2 = 0;
+    int32_t acc_n3 = 0;
+
+    const int8_t *ip_row_0 = row_base;
+    const int8_t *ip_row_1 = row_base + row_elements;
+    const int8_t *ip_row_2 = row_base + (2 * row_elements);
+    const int8_t *ip_row_3 = row_base + (3 * row_elements);
+    int32_t sum_tmp = 0;
+
+#if defined(ARM_MATH_MVEI)
```

Comment:
You probably need another flag for the future since it will have to also compile on other compilers. And I am not sure the embedded asm syntax is always the same for all compilers.

So, I think it would be safer to have another compilation flag to select the ASM version or the intrinsic version.

Reply:
@christophe0606 all right.. i'll add one in the next patch..